src/tactic/lint/type_classes.lean

```diff
@@ -20,7 +20,9 @@ and the appropriate definition of instances:
  * `fails_quickly` checks that type class resolution finishes quickly
  * `has_coe_variable` checks that there is no instance of type `has_coe α t`
  * `inhabited_nonempty` checks whether `[inhabited α]` arguments could be generalized
-   to `[inhabited α]`
+   to `[nonempty α]`
+ * `decidable_classical` checks whether `[decidable_... p]` arguments do not show up in the
+   statement, and could thus be removed by going classical in the proof.
```

Comment:
```suggestion
 * `decidable_classical` checks propositions for `[decidable_... p]` arguments that are not used in the statement, and could thus be removed by using `classical` in the proof.
```

Mostly I wanted to indicate that this linter only applies to propositional declarations.